sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -136,6 +139,27 @@ def add(self, other: Pipeline) -> None:
         for source, dests in other.outgoing_edges.items():
             self.outgoing_edges[source] = dests
 
+    def start(self, step: Source) -> Pipeline:
+        assert not self.__previous_step, "Cannot start a pipeline with multiple sources"
+        step.register(self)
+        self.__previous_step = step
+        return self
+
+    def apply(self, step: Step) -> Pipeline:
+        assert self.__previous_step, "Cannot apply a step without starting first"
+        step.register(self, self.__previous_step)
```

Comment:
I am not sure I get why we need to provide the step a reference to the pipeline.